src/include/encoding.h

```diff
@@ -280,7 +280,8 @@ inline void decode_nohead(int len, bufferlist& s, bufferlist::iterator& p)
 
 // Time, since the templates are defined in std::chrono
 
-template<typename Clock, typename Duration>
+template<typename Clock, typename Duration,
+         typename std::enable_if_t<!Clock::is_steady>* = nullptr>
```

Comment:
This bothers me only because it's inaccurate. The thing that makes mono_clock un-serializable isn't that it's steady, it's that it has no epoch in common with anything else. It would probably be better in the long run to make this test contingent on whether it has `to_time_t` or `to_timespec`. (denc.h has an example of how to make templates that check for the existence of functions.

Though maybe don't worry about it. I think the 'clock' classes are a bit overlarge, I might just do a refactor sometime and make most of the conversions free functions that depend on to_time_t and from_time_t to do their thing.

Reply:
okay, thanks. i'll switch back to using void_t to detect the helper functions needed. then the predicate won't be misleading, and it won't erroneously match std::chrono::system_clock (but fail to compile due to lack of helper functions)